### PR TITLE
feat(FEC-10057): move the plugin manager to kaltura player

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

* Remove plugin management:
   - `PluginManager` and `BasePlugin`
   - plugin types and interfaces:
   `PKPluginsConfigObject `, `IMiddlewareProvider`, `IPlugin`and `IUIComponentsProvider `
 
* Remove ads management:
   - `Ad` and `adBreak`
   - `AdsController` and `ControllerProvider`
   - ad types and interfaces:
   `PKAdBreakObject`, `PKAdPod`, `PKAdObject`, `PKAdvertisingConfigObject`, `IAdsController`, `IAdsControllerProvider` and `IAdsPluginController`
   - advertising and waterfalling management

* Remove `PKUIComponent` and `IUIComponentsProvider`

* Expose `EngineDecorator` registry and  the `playbackMiddleware` 
* Remove the VR plugin API

* Remove `PLAYBACK_START` and `PLAYBACK_ENDED` firing and handling:
    - The [fix](https://github.com/kaltura/playkit-js/pull/405) of [FEC-9333](https://kaltura.atlassian.net/browse/FEC-9333) moved to https://github.com/kaltura/playkit-js-ima-dai/pull/41
    - The [fix](https://github.com/kaltura/playkit-js/pull/440) of [FEC-9534](https://kaltura.atlassian.net/browse/FEC-9534) has been changed to use the `IDLE `state

Solves FEC-10057

Related to https://github.com/kaltura/kaltura-player-js/pull/332

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [x] Docs have been updated

### CODE REVIEW

**Comment by Yuvalke** [comment]
There is an ad events on playkit which will not be raised when the plugin manager on Kaltura player.

File: docs/configuration.md [hunk 1]
```diff
@@ -796,18 +743,6 @@ var config = {
 >
 > ##
 >
-> > ### config.playback.loop
```
Comment (Yuvalke) [inline docs/configuration.md:799]: why did you move loop from playkit to kaltura player?

Comment (yairans) [reply docs/configuration.md:799]: since it affected by the `PLAYBACK_ENDED` event
